### ISSUE TITLE
Flickering/blinking when using sway hide mode and IPC

### ISSUE THREAD

**Issue by mavaa** [Alexays/Waybar#3750]
Not sure if this belongs here or in the sway repo, but after one of the latest sway and/or waybar updates, my bar started flickering/blinking when hidden by sway. I've observed this on multiple machines, both with AMD and NVidia GPUs. I'm using this configuration for sway to have the bar show/hide when I use the win key:

```
bar {
    swaybar_command waybar
    mode hide
    position top

}
```

I've also minified my waybar config to try to identify if anything triggers it. This config at least reproduces it:
```
// -*- mode: jsonc -*-
{
    "spacing": 0, // Gaps between modules (4px)
    "ipc": true,
    "modules-left": [
        "sway/workspaces",
        "sway/mode",
        "sway/scratchpad",
    ],
    "modules-center": [
        "sway/window"
    ],
    "modules-right": [
        "mpris",
        "pulseaudio",
        "memory",
        "temperature",
        "keyboard-state",
        "battery",
        "battery#bat2",
        "clock",
        "tray"
    ],
}
```

Removing the tray item sometimes seems to fix the issue, and stopping nm-applet greatly reduced the flickering, but after restarting sway a few times it sometimes comes back. And even when not flickering, very often a black bar stays where the waybar used to be. It seems almost as if the more stuff I add to my config, the worse it gets. Moving the mouse to the screen with flickering often (but not always) removes it. I tried to capture some video of the issue, but due to framerate and/or the fact that OBS is recording the screen, it doesn't pick up the full effect:

https://github.com/user-attachments/assets/832cc2d4-52bd-42e4-8328-105747547dec

In reality, when looking on my monitor, the black line observed at the top is flickering on and off like crazy, several times per second. I also have a stylesheet with catppuccin theme added in the video above, but the same issue still happened if I removed it.

**Comment by lkp26k** [comment]
Have the same problem. Tried to fix it the last three days. I'm exhausted about it. I'd be very grateful for anything remotely helpful.

`waybar -l debug` shows that the visibility state is switched correctly. I was assuming something in the GTK direction, but that's a blind guess and fiddling there wasn't successful. 

I'm on Arch with Intel/Nvidia (nouveau) GPU. I'm new to this, so I don't know what information is relevant to share - just let me know :)

**Comment by lkp26k** [comment]
`bindsym $mod+c exec pkill -SIGUSR1 waybar`

From what I am seeing, the artifacts don't occur, they remain (after the bar was visible). 

It's an ugly solution, but once you manually set the visibility state, they don't come back. For now, I'll just update the visibility state when I see artifacts. That's the best I can think of right now.

**Comment by wkordalski** [comment]
I have the same issue on NixOS/unstable with AMD integrated graphics.

I've checked that [this](https://github.com/NixOS/nixpkgs/commit/5153b7b1a3f83ff175df9be1ab5b17754519ac67) commit to `nixpkgs` introduces the bug on my hardware.
The commit updates sway (from 1.9 to 1.10) and wlroots (from 0.17 to 0.18).

Maybe this helps someone to fix this issue.

**Comment by SeabertYuan** [comment]
I was struggling with this problem too, and unfortunately I think this is a duplicate of #3492. I had to just accept not hiding my bar for now.

**Comment by lkp26k** [comment]
> I was struggling with this problem too, and unfortunately I think this is a duplicate of #3492. I had to just accept not hiding my bar for now.

Good call. The issue you linked has some more info in it. Maybe it gets fixed anytime soon. 

I'm not giving up any vertical pixel. Switched back to the basic Swaybar. 'Hide' works fine with that.

**Comment by arsmoriendy** [comment]
I have the same issue except it doesn't flicker, just static black.

I ended up using a workaround inspired from #3492 by setting my background color alpha value to a really high number like 0.99 in my gtk css file:

```css
window#waybar {
  background-color: alpha(black, 0.99)
}
```

**Comment by detoxifiedplant** [comment]
i also started having the same issue,
only happens when activating `hide` mode,
have switched to swaybar for temporarily.

**Comment by detoxifiedplant** [comment]
i also started having the same issue,
only happens when activating `hide` mode,
having made the changes suggested in #3492 helped

**Comment by Alexays** [comment]
Closing as a duplicate of #3492 (same sway-hide opaque-buffer flicker). 🙏
